Podcast/ExternalProfileViewController.swift

```diff
@@ -93,47 +104,96 @@ class ExternalProfileViewController: UIViewController, UITableViewDataSource, UI
-        episode.tags = [Tag(name:"Design"), Tag(name:"Learning"), Tag(name: "User Experience"), Tag(name:"Technology"), Tag(name:"Innovation"), Tag(name:"Dogs")]
-        user.favoriteEpisodes = Array(repeating: episode, count: 5)
+        // Now request user subscriptions and favorites
+//        let favoritesRequest = GetUserFavoritesEndpointRequest(userID: id)
+//        favoritesRequest.success = { (favoritesEndpointRequest: EndpointRequest) in
+//            DispatchQueue.main.async {
+//                guard let results = favoritesEndpointRequest.processedResponseValue as? [Episode] else { return }
+//                self.favorites = results
+//                self.profileTableView.reloadData()
+//            }
+//        }
+//        System.endpointRequestQueue.addOperation(favoritesRequest) // UNCOMMENT WHEN FAVORITES ARE DONE
         
-        updateViewWithUser(user)
+        let subscriptionsRequest = FetchUserSubscriptionsEndpointRequest(userID: id)
+        subscriptionsRequest.success = { (subscriptionsEndpointRequest: EndpointRequest) in
+            DispatchQueue.main.async {
+                guard let results = subscriptionsEndpointRequest.processedResponseValue as? [SubscriptionSeries] else { return }
+                self.subscriptions = results
+                self.profileTableView.reloadData()
+            }
+        }
+        System.endpointRequestQueue.addOperation(subscriptionsRequest)
     }
     
     func updateViewWithUser(_ user: User) {
         self.user = user
         // Update views
         profileHeaderView.setUser(user)
         miniHeader.setUser(user)
-        profileTableView.reloadData()
     }
     
     func didPressBackButton() {
-        let _ = navigationController?.popViewController(animated: true)
+        if let navController = navigationController {
+            navController.popViewController(animated: true)
+        }
     }
     
     // Mark: - ProfileHeaderView
     func profileHeaderDidPressFollowButton(profileHeader: ProfileHeaderView, follow: Bool) {
         // Follow/Unfollow someone
+        if follow {
+            let newFollowRequest = NewFollowingEndpointRequest(userID: user.id)
+            newFollowRequest.success = { (endpointRequest: EndpointRequest) in
+                DispatchQueue.main.async {
+                    // Follow succeded
+                    profileHeader.followButton.isSelected = true
+                    self.user.isFollowing = true
+                }
+            }
+            newFollowRequest.failure = { (endpointRequest: EndpointRequest) in
+                DispatchQueue.main.async {
+                    // Follow failed
+                    profileHeader.followButton.isSelected = false
+                    self.user.isFollowing = false
+                }
+            }
+            System.endpointRequestQueue.addOperation(newFollowRequest)
```

Comment:
you should probably have something to prevent user from tapping follow/unfollow multiple times